### PR TITLE
Add checker for bad migration timestamps

### LINKED ISSUE #113: Check for invalid dates in migration names
An invalid date in a migration file name is considered a valid value, but then migrations take the form of numerics instead of dates from then on causing weirdness.  This check would prevent that.

### PR DESCRIPTION
**Pull request by Fryguy**
Resolves #113 

@jrafanie Please review.

### CODE REVIEW

File: app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require 'time'
+
+module CommitMonitorHandlers::Batch
+  class GithubPrCommenter::MigrationDateChecker
```
Comment (jrafanie) [inline app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb]: Is there a reason to use this condensed style?  It seems to be more hassle with autoload and constants on each namespace than useful.

File: app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    def valid_timestamp?(ts)
+      Time.parse(ts)
+    rescue ArgumentError
+      false
+    end
+
+    def diff_files_for_commit_range
+      branch.repo.with_git_service do |git|
+        git.diff_details(*commit_range).keys
```
Comment (jrafanie) [inline app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb]: I like this, does it make sense to have a helper method for all commit/commit range handlers that gets the diff_details based on the commit/commit range?  It seems like all of them might want to look at the commit data.

**Comment by jrafanie** [comment]
LGTM, other than the namespacing question.

File: app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb (hunk 1 above)
Comment (Fryguy) [inline app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb]: I didn't want 5 levels of indent...it's all going to change anyway when I drop the "type" part

File: app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb (hunk 2 above)
Comment (Fryguy) [inline app/workers/commit_monitor_handlers/batch/github_pr_commenter/migration_date_checker.rb]: Yes I plan to in the future...wanted a few in place so a pattern could emerge. https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

**Pull request merged**